source/backtrace/backtrace.d

```diff
@@ -173,11 +188,32 @@ private void printPrettyTrace(const(void*[]) bt, File output, PrintOptions optio
     else return "\u001B[0m";
   }
 
-  output.writeln("Stack trace:");
+  output.put("Stack trace:");
+  if (insertNewlines) output.put("\n");
 
   foreach(i, t; trace.drop(framesToSkip)) {
     auto symbol = symbols[framesToSkip + i].demangled;
-    output.writeln("#", i + 1, ": ", forecolor(Color.red), t.file, reset(), " line ", forecolor(Color.yellow), "(", t.line, ")", reset(), symbol.length == 0 ? "" : " in ", forecolor(Color.green), symbol, reset(), " @ ", forecolor(Color.green), "0x", bt[i + 1], reset());
+    auto s = appender!string();
+    formattedWrite(
```

Comment:
Try putting `output` directly in as first argument to `formattedWrite`. If that doesn't work, take the appender out of the loop.